pkg/handlers/archive.go

```diff
@@ -115,6 +119,33 @@ func (a *Archive) openArchive(ctx logContext.Context, depth int, reader io.Reade
 }
 
 func (a *Archive) handleNonArchiveContent(ctx logContext.Context, reader io.Reader, archiveChan chan []byte) error {
+	if a.skipBinaries {
+		buffer := make([]byte, 512)
+		n, err := reader.Read(buffer)
+		if err != nil && !errors.Is(err, io.EOF) {
+			return fmt.Errorf("unable to read file for MIME type detection: %w", err)
+		}
+
+		// Create a new reader that starts with the buffer we just read
+		// and continues with the rest of the original reader.
+		reader = io.MultiReader(bytes.NewReader(buffer[:n]), reader)
+
+		mime := mimetype.Detect(buffer)
+		mimeT := mimeType(mime.String())
+
+		fmt.Println(mimeT)
+
+		if mimeT == machOType || mimeT == octetStream {
+			ctx.Logger().V(5).Info("skipping binary file", "ext", mimeT)
+			return nil
+		}
+
+		if common.SkipFile(mime.Extension()) {
+			ctx.Logger().V(5).Info("skipping file", "ext", mimeT)
+			return nil
+		}
```

Comment:
Why do we do this only if `skipBinaries` is set?

Reply:
Oh i forgot, we had added a bunch of binary extensions to that function, but that probably doesn't make much sense anymore. I'll update that function to remove binary extensions.